browser/net/ipfs_redirect_network_delegate_helper.cc

```diff
@@ -42,4 +44,36 @@ int OnBeforeURLRequest_IPFSRedirectWork(
   return net::OK;
 }
 
+int OnHeadersReceived_IPFSRedirectWork(
+    const net::HttpResponseHeaders* response_headers,
+    scoped_refptr<net::HttpResponseHeaders>* override_response_headers,
+    GURL* allowed_unsafe_redirect_url,
+    const brave::ResponseCallback& next_callback,
+    std::shared_ptr<brave::BraveRequestInfo> ctx) {
+  if (!ctx->browser_context ||
+      IsIpfsResolveMethodDisabled(ctx->browser_context)) {
+    return net::OK;
+  }
+
+  std::string ipfs_path;
+  if (ctx->ipfs_auto_fallback && response_headers &&
+      response_headers->GetNormalizedHeader("x-ipfs-path", &ipfs_path) &&
+      // Make sure we don't infinite redirect
+      !ctx->request_url.DomainIs(ctx->ipfs_gateway_url.host())) {
+    GURL::Replacements replacements;
+    replacements.SetPathStr(ipfs_path);
+    GURL new_url = ctx->ipfs_gateway_url.ReplaceComponents(replacements);
+
+    *override_response_headers =
+        new net::HttpResponseHeaders(response_headers->raw_headers());
+    (*override_response_headers)
+        ->ReplaceStatusLine("HTTP/1.1 307 Temporary Redirect");
+    (*override_response_headers)->RemoveHeader("Location");
+    (*override_response_headers)->AddHeader("Location", new_url.spec());
```

Comment:
these two lines are basically `SetHeader`

Reply:
Since this is already merged, I'll do that in a follow-up.